cloudflare/resource_cloudflare_notification_policy.go

```diff
@@ -155,7 +160,7 @@ func buildNotificationPolicy(d *schema.ResourceData) cloudflare.NotificationPoli
 	}
 
 	if filters, ok := d.GetOk("filters"); ok {
-		notificationPolicy.Filters = filters.(map[string][]string)
+		notificationPolicy.Filters = filters.([]interface{})[0].(map[string][]string)
```

Comment:
if i'm understanding this correctly, you'll need to build this `map[string][]string` instead of casting it. looping over `filters` and appending should do it.